sriov/tests/SR_IOV_macAddressDuplicate/test_SR_IOV_macAddressDuplicate.py

```diff
@@ -0,0 +1,58 @@
+from sriov.common.utils import create_vfs, execute_and_assert
+
+
+def test_SR_IOV_macAddressDuplicate(dut, settings):
+    """Test and ensure that duplicate mac address across VFs on the same PF is
+       permitted
+
+    Args:
+        dut:      ssh connection obj
+        settings: settings obj
+    """
+
+    mac_1 = "aa:bb:cc:dd:ee:11"
+    mac_2 = "aa:bb:cc:dd:ee:22"
+    pf = settings.config["dut"]["interface"]["pf1"]["name"]
+    steps = [
+        f"ip link set {pf} vf 0 mac {mac_1}",
+        f"ip link set {pf} vf 1 mac {mac_2}",
+    ]
+
+    create_vfs(dut, pf, 2)
+
+    execute_and_assert(dut, steps, 0)
+
+    # check if vf 0 mac address is equal to mac_1
+    cmd = ["ip link show {}".format(pf) + " | awk '/vf 0/{print $4;}'"]
+    outs, errs = execute_and_assert(dut, cmd, 0)
+    vf_mac = outs[0][0].strip("\n")
+    print(vf_mac)
+    assert vf_mac == mac_1
+
+    # check if vf 1 mac address is equal to mac_2
+    cmd = ["ip link show {}".format(pf) + " | awk '/vf 1/{print $4;}'"]
```

Comment:
this code block was repeated 3 times; I wonder if we should use a macro for the repeated steps. For bonding I have the similar situation - the validation code are common steps for both DPDK and non DPDK test case, so I created a macros.py for the collection of steps that are common

Reply:
I found even better option, using verify_vf_address for that check.